In [1]:
import pandas as pd

df = pd.read_csv("../data/raw/ev_sample_50k.csv")

print("Shape:", df.shape)
print("\nColumn data types:\n", df.dtypes)
print("\nMissing values per column:\n", df.isna().sum())
print("\nDuplicate rows:", df.duplicated().sum())

Shape: (50000, 16)

Column data types:
 VIN (1-10)                                            object
County                                                object
City                                                  object
State                                                 object
Postal Code                                          float64
Model Year                                             int64
Make                                                  object
Model                                                 object
Electric Vehicle Type                                 object
Clean Alternative Fuel Vehicle (CAFV) Eligibility     object
Electric Range                                       float64
Legislative District                                 float64
DOL Vehicle ID                                         int64
Vehicle Location                                      object
Electric Utility                                      object
2020 GEOID                                   

In [2]:
print("Rows with Electric Range = 0:", (df["Electric Range"] == 0).sum())
print("\nState value counts:\n", df["State"].value_counts())
print("\nBase MSRP column exists:", "Base MSRP" in df.columns)

Rows with Electric Range = 0: 32752

State value counts:
 State
WA    49873
CA       30
VA       19
OR        9
FL        9
MD        7
NC        6
TX        5
LA        3
NY        3
GA        3
AZ        3
CO        3
PA        2
ID        2
NJ        2
HI        2
MO        2
BC        2
IL        2
IN        2
MN        1
CT        1
DC        1
NM        1
KY        1
UT        1
MA        1
OH        1
GU        1
NH        1
TN        1
Name: count, dtype: int64

Base MSRP column exists: False


In [3]:
print("Duplicate VINs:", df["VIN (1-10)"].duplicated().sum())
print("\nCAFV Eligibility values:\n", df["Clean Alternative Fuel Vehicle (CAFV) Eligibility"].value_counts())

Duplicate VINs: 38907

CAFV Eligibility values:
 Clean Alternative Fuel Vehicle (CAFV) Eligibility
Eligibility unknown as battery range has not been researched    32752
Clean Alternative Fuel Vehicle Eligible                         13160
Not eligible due to low battery range                            4088
Name: count, dtype: int64


Quality findings:

Duplicates: 0
Electric Range = 0.     : 32,752 rows (65.5%) — confirmed = "not researched," not real zero. Main issue, use for sensitivity analysis
Legislative District.   : 128 missing (largest gap)
Minor missing (2–4 rows): County, City, Postal Code, Vehicle Location, Electric Utility, 2020 GEOID
Out-of-state            : 127 rows (0.25%)
VIN(1-10) = make/model/trim, not unique ID, shared values are normal